src/sdk/namespace_fs.js

```diff
@@ -1709,9 +1741,72 @@ class NamespaceFS {
     // OBJECT RESTORE //
     ////////////////////
 
+    /**
+     * restore_object restores simply sets teh restore request xattr
+     * which should be picked by another mechanism.
+     * 
+     * restore_object internally relies on 3 xattrs:
+     * - XATTR_RESTORE_REQUEST
+     * - XATTR_RESTORE_ONGOING
+     * - XATTR_RESTORE_EXPIRY
+     * @param {*} params 
+     * @param {nb.ObjectSDK} object_sdk 
+     * @returns {Promise<undefined>}
+     */
     async restore_object(params, object_sdk) {
-        dbg.log0('restore_object', params);
-        throw new Error('TODO');
+        dbg.log0('restore_object:', params);
+
+        if (!await this._is_storage_class_supported(s3_utils.STORAGE_CLASS_GLACIER)) {
```

Comment:
passing s3_utils.STORAGE_CLASS_GLACIER to this._is_storage_class_supported() is by mistake?

Reply:
Actually it tests if the deployment supports `GLACIER` storage class or not (I don't think `GLACIER IR` needs `restore-object` either).